test/upgrading/upgrader.spec.ts

```diff
@@ -742,7 +757,7 @@ describe('libp2p.upgrader', () => {
     expect(streamCount).to.equal(1)
 
     await expect(localToRemote.newStream(protocol)).to.eventually.be.rejected()
-      .with.property('code', 'ERR_STREAM_RESET')
+      .with.property('code', 'ERR_UNDER_READ')
```

Comment:
Would this now be the expected error code?

Reply:
Yamux should behave the same as Mplex so the expected error code is the one in the test - `ERR_STREAM_RESET`

Reply:
So it seems that according to the test,   _it should limit the number of incoming streams that can be opened using a protocol_ does this mean the error below is correct?

```bash
     Error: stream ended before 1 bytes became available
```

The error is occurring in [it-reader](https://github.com/alanshaw/it-reader/blob/master/src/index.ts#L71) which is a dep of [multistream-select](https://github.com/libp2p/js-libp2p-multistream-select/blob/master/package.json#L155), it happens when trying to [read the protocol response](https://github.com/libp2p/js-libp2p-multistream-select/blob/master/src/select.ts#L81)

Reply:
It seems that in js-libp2p-yamux [the `resetController` isn't an abortable source](https://github.com/ChainSafe/js-libp2p-yamux/blob/da397130d09fda8c1b11196128985efaacfe6f6b/src/stream.ts#L129) and so my inclination is that when the [resetController is aborted](https://github.com/libp2p/js-libp2p-mplex/blob/31a32f386f582467b4cda578a823610f4e096a5e/src/stream.ts#L191) the `ERR_STREAM_RESET` is **not** propagated but rather this [lower level error is logged](https://github.com/ChainSafe/js-libp2p-yamux/blob/da397130d09fda8c1b11196128985efaacfe6f6b/src/stream.ts#L146), I'll open a PR